[BE-4051]: All permissions have been changed and added detailed tables (#552)
appcircleio/appcircle-docusaurus: docs/account/my-organization/role-management.md

  ]
---

# Role Management

> With Appcircle's [**Advanced Role Management**](/account/my-organization#advanced-role-management) structure, you can detail the roles of the members you add to your organisation separately for each module and restrict the authority of the members. Appcircle offers different role types for each module. A short description of each role is given in the table below. In addition, you can find detailed role management information for each module under module titles.

Reviewer: ```suggestion
With Appcircle's [Advanced Role Management](/account/my-organization#advanced-role-management) structure, you can assign specific roles to organization members for each module, allowing you to manage and restrict their permissions effectively. Appcircle provides various role types for each module, with a brief description of each role provided in the table below. For more detailed information on role management for each module, please refer to the respective module titles.
```

> - **Owner**: The user is authorised unlimited access to all modules.

Reviewer: ```suggestion
- **Owner**: The user is authorized for unlimited access to all modules.
```

- **Manager**: The user becomes the administrator of the relevant module and no restrictions are made.

> - **Operator**: The user is the person who manages the operations of the relevant module and certain restrictions apply.

Reviewer: ```suggestion
- **Operator**: The user manages the operations of the relevant module, with certain restrictions in place.
```

- **Ext. Operator**: The user has very limited authorisation in the relevant module. It is usually for third party employees from outside the company.

> - **Viewer**: User only has view authorisation in the relevant module, cannot take any action.

Reviewer: ```suggestion
- **Viewer**: The user only has view authorization in the relevant module and cannot take any action.
```

:::caution Role Types

> Some role types are not used in some modules because they are not required in that module because they are the same as another role. Therefore, roles may differ for each module.

Reviewer: ```suggestion
Some role types are not used in certain modules because they are redundant or unnecessary, as they serve the same function as another role. Therefore, roles may vary for each module.
```

:::

### Build Permissions

> The following table details the roles and restrictions for the [**Build**](/build) module. Please refer to the modules related information and coution notes. 

Reviewer: ```suggestion
The following table details the roles and restrictions for the [**Build**](/build) module. Please refer to the related module information and coution notes. 
```

| Build Actions       | Delete Commit Artifacts                    | ✅     | ✅       | ⛔        | ⛔      |
| Build Actions       | Download Artifacts                         | ✅     | ✅       | ✅        | ✅      |
| Build Actions       | Distribution Binary                        | ✅     | ✅       | ✅        | ⛔      |
| Test Results        | List Test Results                          | ✅     | ✅       | ✅        | ✅      |
| Connection          | Add/Delete/Update Connections (User Based) | ✅     | ✅       | ✅        | ✅      |
| Connection          | List Connection (User Based)               | ✅     | ✅       | ✅        | ✅      |
| Runner              | Add/Delete/Update Runner(Root Only)        | ✅     | ⛔       | ⛔        | ⛔      |
| Runner              | List Runner(Root Only)                     | ✅     | ✅       | ✅        | ✅      |
| Runner Access Token | Create/Delete Runner Access Token          | ✅     | ⛔       | ⛔        | ⛔      |
| Runner Access Token | List Runner Access Token                   | ✅     | ⛔       | ⛔        | ⛔      |
| Report              | List Build Reports                         | ✅     | ✅       | ✅        | ✅      |

> :::caution Distrubition Binary and Runner Details

Reviewer: ```suggestion
:::caution Distribution Binary and Runner Details
```

:::

### Environment Variables Permissions

The following table details the roles and restrictions for the [**Environment Variables**](/environment-variables). Please refer to the modules related information and coution notes. 

| Environment Variable | Scopes                                        | Owner | Manager | Viewer |
|----------------------|-----------------------------------------------|-------|---------|--------|
| Environment Variable | Add/Delete/Update Environment Variable Groups | ✅     | ✅       | ⛔      |
| Environment Variable | Add/Delete/Update Environment Variable        | ✅     | ✅       | ⛔      |
| Environment Variable | List Environment Variable                     | ✅     | ✅       | ✅      |
| Environment Variable | List Environment Variable Groups              | ✅     | ✅       | ✅      |

:::info

**Manager**, **Operator** and **Viewer** Environment Variable permissions can use variable groups in [**Build profile configuration**](/build/build-process-management/build-profile-configuration#environment-variables-configuration).

:::

### Signing and Identity Permissions

> The following table details the roles and restrictions for the [**Signing and Identity**](/signing-identities) module. Please refer to the modules related information and coution notes. 

Reviewer: ```suggestion
The following table details the roles and restrictions for the [**Signing and Identity**](/signing-identities) module. Please refer to the related module information and coution notes. 
```

| Apple Cerficate                 | List Apple Certificates                | ✅     | ✅       | ✅      |
| Apple Cerficate Signing Request | Add/Delete/Download CSR                | ✅     | ✅       | ⛔      |
| Apple Cerficate Signing Request | Convert CSR to .p12                    | ✅     | ✅       | ⛔      |
| Apple Cerficate Signing Request | List CSR                               | ✅     | ✅       | ✅      |
| Apple Identifiers               | Add/Delete/Update Apple Identifiers    | ✅     | ✅       | ⛔      |
| Apple Identifiers               | List Apple Identifiers                 | ✅     | ✅       | ✅      |
| Apple Profile                   | Add/Delete/Update Apple Profiles       | ✅     | ✅       | ⛔      |
| Apple Profile                   | List Apple Profiles                    | ✅     | ✅       | ✅      |
| Keystore                        | Add/Delete/Update Keystores            | ✅     | ✅       | ⛔      |
| Keystore                        | List Keystores                         | ✅     | ✅       | ✅      |
| Report                          | List Signing Reports                   | ✅     | ✅       | ✅      |

:::info Signing and Identities

> **Manager**, **Operator** and **Viewer** Signing Identity Management permissions can use signing identities in [**Build profile configuration**](/build/build-process-management/build-profile-configuration#environment-variables-configuration).

Reviewer: ```suggestion
**Manager** and **Viewer** Signing Identity Management permissions can use signing identities in [**Build profile configuration**](/build/build-process-management/build-profile-configuration#environment-variables-configuration).
```

:::

### Testing Distribution Permissions

> The following table details the roles and restrictions for the [**Testing Distribution**](/distribute) module. Please refer to the modules related information and coution notes. 

Reviewer: ```suggestion
The following table details the roles and restrictions for the [**Testing Distribution**](/distribute) module. Please refer to the related module information and caution notes. 
```

| Apple Device         | Add/Delete Apple Device                | ✅     | ✅       | ⛔        | ⛔             | ⛔      |
| Apple Device         | Register Apple Device                  | ✅     | ✅       | ⛔        | ⛔             | ⛔      |
| Apple Device         | Adding New Device to Provision         | ✅     | ✅       | ⛔        | ⛔             | ⛔      |
| Apple Device         | Sync from Apple Developer              | ✅     | ✅       | ✅        | ⛔             | ⛔      |
| Apple Device         | List Apple Device                      | ✅     | ✅       | ✅        | ⛔             | ✅      |
| Report               | List Reports App Version               | ✅     | ✅       | ✅        | ⛔             | ✅      |
| Report               | List Reports App Sharing               | ✅     | ✅       | ✅        | ⛔             | ✅      |

:::caution Authentication Settings

If the selected Authentication type is Static login, Manager role can change **Username** and **Password**. However, it cannot change the content for other Authentication types such as **LDAP** or **SSO**.

> 

Reviewer: Need to add a note for pointing out that manager and operator distribution roles need to have a manager signing identity role in order to sync apple device from apple developer.

:::

:::caution Sending Binary

- **Manager** or **Operator** Distribution Profile permission can send to enterprise app store if user has Manager, Operator or Ext. Operator Enterprise App Store permission.
- **Manager** or **Operator** Distribution Profile permission can send to publish if user has Manager or Operator Publish Android and Manager or Operator iOS permission.
- **Manager** or **Operator** Distribution Profile permission can resign binary if user has Manager or Viewer Signing Identity Management permission.

> 

Reviewer: Our notes say that ex operator enterprise role should not be enough to be able to send binary to enterprise from distribution. Might need to remove the ex operator text from there.

:::

### Testing Group Permissions

> The following table details the roles and restrictions for the [**Testing Groups**](/distribute/testing-management/testing-groups). Please refer to the modules related information and coution notes. 

Reviewer: ```suggestion
The following table details the roles and restrictions for the [**Testing Groups**](/distribute/testing-management/testing-groups). Please refer to the related module information and caution notes. 
```

| Testing Groups | Scopes                                  | Owner | Manager | Viewer |
|----------------|-----------------------------------------|-------|---------|--------|
| Testing Groups | Add/Delete/Update Testing Group         | ✅     | ✅       | ⛔      |
| Testing Groups | Add/Delete/Update Testing Group Testers | ✅     | ✅       | ⛔      |
| Testing Groups | List Testing Groups                     | ✅     | ✅       | ✅      |
| Testing Groups | List Testing Group Testers              | ✅     | ✅       | ✅      |

### Publish Module iOS Permissions

> The following table details the roles and restrictions for the [**Publish**](/publish-module) module for iOS. Please refer to the modules related information and coution notes.

Reviewer: ```suggestion
The following table details the roles and restrictions for the [**Publish**](/publish-module) module for iOS. Please refer to the related module information and caution notes.
```

| Organization and Team Management    | Assign Role for User                  | ✅     | ✅       |
| Organization and Team Management    | List User                             | ✅     | ✅       |
| SSO                                 | Create/Delete/Update SSO              | ✅     | ⛔       |
| SSO                                 | List SSO                              | ✅     | ✅       |
| LDAP                                | Add/Delete/Update LDAP                | ✅     | ⛔       |
| LDAP                                | List LDAP                             | ✅     | ✅       |
| PAT                                 | Generate PAT                          | ✅     | ✅       |
| PAT                                 | View PAT                              | ✅     | ✅       |
| Runner Access Token                 | List Runner Access Token              | ✅     | ⛔       |
| Runner Access Token                 | Create/Delete Runner Access Token     | ✅     | ⛔       |
| Report                              | View Organziation Report              | ✅     | ✅       |

:::info Organization Management

> Whatever role a user is assigned in the root organisation, user has the same role in the **sub-organisations**. For example, someone who is a Manager in the root organisation is automatically assigned as Manager in the sub-organisations. 

Reviewer: ```suggestion
Whatever role a user is assigned in the root organization, they will have the same role in the **sub-organizations**. For example, someone who is a Manager in the root organization is automatically assigned as a Manager in the sub-organizations. 
```

> If you want to assign a role in a sub-organisation, please do so within the respective **sub-organisation**.

Reviewer: ```suggestion
If you want to assign a role in a sub-organization, please do so within the respective **sub-organization**.
```

:::

:::caution Organization Management Role Assignment

> The Manager role **cannot** assign itself and another user as **Owner** when assigning the role.

Reviewer: ```suggestion
The Manager role **cannot** assign itself and another user as **Owner** when assigning roles.
```

:::

### Billing Management Permissions

Manage the subscription, payment details, and invoices.

> The following table details the roles and restrictions for the **Billing** details. Please refer to the modules related information and coution notes.

Reviewer: ```suggestion
The following table details the roles and restrictions for the **Billing** details. Please refer to the related module information and caution notes.
```

| Billing Sub-modules | Scopes                    | Owner | Manager |
|---------------------|---------------------------|-------|---------|
| Subscription        | List Subscription Details | ✅     | ✅       |

### Connection Managements

> Connect or disconnect from third-party service providers such as notification tools or store connections

Reviewer: ```suggestion
Connect or disconnect from third-party service providers such as notification tools or store connections.
```
